File: oauth-service/src/main/java/a/gleb/oauthservice/db/repository/AccountRepository.java
package a.gleb.oauthservice.db.repository;

import a.gleb.oauthservice.db.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface AccountRepository extends JpaRepository<Account, UUID> {
    Optional<Account> findAccountByUsernameOrEmail(String username, String email);
}

File: oauth-security-server/src/main/kotlin/a/gleb/oauthsecurityserver/db/repository/AccountRepository.kt
package a.gleb.oauthsecurityserver.db.repository

import a.gleb.oauthsecurityserver.db.entity.Account
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface AccountRepository : JpaRepository<Account, UUID> {
    fun findAccountByUsernameOrEmail(username: String, email: String): Optional<Account>
}